src/transformStateWithClones.js

```diff
@@ -6,8 +6,36 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArr = [];
+
+  for (const item in actions) {
+    newArr[item] = {};
+  }
+  newArr[0] = { ...state };
```

Comment:
```suggestion
  const stateCopy = { ...state };
```
Якось дуже все ускладнено... Можна просто зробити копію стейта, робити всі операції з нею, а потім пушити її в newArr на кожній ітерації в циклі
```
newArr.push({ ...stateCopy })